### PR TITLE
Fix unstable comment just before or-patterns

### PR DESCRIPTION
**Pull request by Julow**
Fixes https://github.com/ocaml-ppx/ocamlformat/issues/1164

### CODE REVIEW

File: lib/Fmt_ast.ml [hunk 1]
```diff
@@ -1030,8 +1030,11 @@ and fmt_pattern c ?pro ?parens ({ctx= ctx0; ast= pat} as xpat) =
                   let loc = xpat.ast.ppat_loc in
                   let force_break = Cmts.has_before c.cmts loc in
                   let leading_cmt =
-                    Cmts.fmt_before ~pro:(Fmt.break 1000 0) ~adj:noop c loc
-                      ~eol:noop
+                    let pro, adj =
+                      if first_grp && first then (noop, fmt "@ ")
+                      else (fmt "@ ", noop)
```
Comment (Julow) [inline lib/Fmt_ast.ml]: I didn't find a case where `break 1000 0` was necessary.

File: lib/Fmt_ast.ml [hunk 2]
```diff
@@ -1030,8 +1030,11 @@ and fmt_pattern c ?pro ?parens ({ctx= ctx0; ast= pat} as xpat) =
                   let loc = xpat.ast.ppat_loc in
                   let force_break = Cmts.has_before c.cmts loc in
                   let leading_cmt =
-                    Cmts.fmt_before ~pro:(Fmt.break 1000 0) ~adj:noop c loc
-                      ~eol:noop
+                    let pro, adj =
+                      if first_grp && first then (noop, fmt "@ ")
```
Comment (Julow) [inline lib/Fmt_ast.ml]: `pro` must be `noop` when `adj` is not to avoid extra empty lines.

**Comment by Julow** [comment]
The last two indeed look like a fix. I'm not 100% sure for the first two, I added a test for that.

**Comment by gpetiot** [comment]
I doubted at first because it kinda looks weird since the comment is long, no doubt for a shorter comment. But this is what is expected for `break-cases=fit`.

**Comment by Julow** [comment]
I see. I tested too and it's a fix.
Should I put that in the changelog ? If yes, how ?

**Comment by gpetiot** [comment]
If you don't find this too long I would modify your entry: `Fix an unstable comment bug in or-patterns; and fix wrapping of or-patterns in presence of comments when break-cases=fit`

**Comment by Julow** [comment]
That's a bit long for a single line, I tried something.

**Pull request merged**